core/src/test/scala/org/apache/spark/shuffle/sort/SortShuffleWriterSuite.scala

```diff
@@ -0,0 +1,98 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.shuffle.sort
+
+import org.mockito.Mockito._
+import org.mockito.MockitoAnnotations
+import org.scalatest.Matchers
+
+import org.apache.spark.{Partitioner, SharedSparkContext, ShuffleDependency, SparkFunSuite}
+import org.apache.spark.memory.MemoryTestingUtils
+import org.apache.spark.serializer.JavaSerializer
+import org.apache.spark.shuffle.{BaseShuffleHandle, IndexShuffleBlockResolver}
+import org.apache.spark.util.Utils
+
+class SortShuffleWriterSuite extends SparkFunSuite with SharedSparkContext with Matchers {
+
+  private val shuffleId = 0
+  private val numMaps = 5
+  private var shuffleHandle: BaseShuffleHandle[Int, Int, Int] = _
+  private val shuffleBlockResolver = new IndexShuffleBlockResolver(conf)
+  private val serializer = new JavaSerializer(conf)
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    MockitoAnnotations.initMocks(this)
+    val partitioner = new Partitioner() {
+      def numPartitions = numMaps
+      def getPartition(key: Any) = Utils.nonNegativeMod(key.hashCode, numPartitions)
+    }
+    shuffleHandle = {
+      val dependency = mock(classOf[ShuffleDependency[Int, Int, Int]])
+      when(dependency.partitioner).thenReturn(partitioner)
+      when(dependency.serializer).thenReturn(serializer)
+      when(dependency.aggregator).thenReturn(None)
+      when(dependency.keyOrdering).thenReturn(None)
+      new BaseShuffleHandle(shuffleId, numMaps = numMaps, dependency)
+    }
+  }
+
+  override def afterAll(): Unit = {
+    try {
+      shuffleBlockResolver.stop()
+    } finally {
+      super.afterAll()
+    }
+  }
+
+  test("write empty iterator") {
+    val context = MemoryTestingUtils.fakeTaskContext(sc.env)
+    val writer = new SortShuffleWriter[Int, Int, Int](
+      shuffleBlockResolver,
+      shuffleHandle,
+      mapId = 1,
+      context
+    )
+    writer.write(Iterator.empty)
+    writer.stop(success = true)
+    val dataFile = shuffleBlockResolver.getDataFile(shuffleId, 1)
+    assert(!dataFile.exists())
+    assert(dataFile.length() === 0)
+    assert(context.taskMetrics().shuffleWriteMetrics.bytesWritten === 0)
+    assert(context.taskMetrics().shuffleWriteMetrics.recordsWritten === 0)
+
+  }
+
+  test("write with some records") {
+    val context = MemoryTestingUtils.fakeTaskContext(sc.env)
+    val records = Iterator((1, 2), (2, 3), (4, 4), (6, 5))
+    val writer = new SortShuffleWriter[Int, Int, Int](
+      shuffleBlockResolver,
+      shuffleHandle,
+      mapId = 2,
+      context
+    )
+    writer.write(records)
+    writer.stop(success = true)
+    val dataFile = shuffleBlockResolver.getDataFile(shuffleId, 2)
```

Comment:
Delete the `dataFile` at the end of the test.

Reply:
@wangjiaochun you are right that DiskBlockManager deletes the directory. 
So please remove `Utils.deleteRecursively`.

So only the `Can we compare the certain expected value instead of != 0 in these three lines?` question left at https://github.com/apache/spark/pull/24080#discussion_r268482573